packages/flutter/lib/src/material/colors.dart

```diff
@@ -88,6 +90,75 @@ class MaterialAccentColor extends ColorSwatch<int> {
   Color get shade700 => this[700];
 }
 
+/// Signature for the function that returns a (potentially different) color
+/// based on a given set of states.
+typedef ColorFunction = Color Function(Set<MaterialState> states);
+
+/// Defines a [Color] that can react to changes in the state of a Material
+/// component, based on a given set of [MaterialState]s.
+///
+/// This is useful for preserving the accessibility of text in different states
+/// of a component. For example, in a [TextButton] with blue text, the text will
```

Comment:
TextButton should be FlatButton here and below

Reply:
Right, my bad.